### DISCUSSION TITLE
Dealing with non-square images in reverse image search

### DISCUSSION THREAD

**Discussion by ai4ki** [milvus-io/milvus#22237] [Q&A and General discussion]
A lot of DL models in computer vision like ResNet, EfficientNet or CLIP need a square input image (like 224 x 224). That's why their preprocessors center-crop any non-square input image accordingly. While the corresponding loss of information may not be a problem for certain applications, it obviously can be for reverse image search (imagine you want to make a search engine for pencils). 

To deal with this problem, I can only think of padding or skewing non-square input images, which might work for some image types but rather not for others. Is there a more thorough way to go about this? Are there any studies evaluating different approaches in terms of suitable metrics like mAP?

I'm not sure if this topic belongs here, but I'd be super grateful for any pointers!

**Comment by xiaofan-luan** [comment]
you might want to see https://github.com/towhee-io/towhee? 
it helps you to do all the preprocessing stage and get your embedding

**Reply by ai4ki** [reply]
Thanks, the towhee repo is a very helpful resource but my problem is more specific. I know what the standard preprocessing stage does and how to get embeddings. I'm looking for solutions in semantic search which deal with non-square images when a model expects a square one as input.
